### PR TITLE
Add Hide Groups plugin

### PR DESCRIPTION
**Pull request by KingRainbow44**
This plugin allows users to hide group DMs via the context menu.

### CODE REVIEW

File: src/plugins/hideGroups/index.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+const hiddenGroupsStore = createStore("HiddenGroups", "HiddenGroupsList");
+
+const hiddenGroups: string[] = [];
+const elements: { [key: string]: HTMLElement } = {};
+
+async function saveToStorage(): Promise<void> {
+    await set("list", JSON.stringify(hiddenGroups), hiddenGroupsStore);
+}
+
+function onChange(): void {
```
Comment (MrDiamondDog) [inline src/plugins/hideGroups/index.tsx:37]: Avoid DOM manipulation when patches are available.

**Review by MrDiamondDog — COMMENTED** [review]
I feel like this should be able to hide normal dms as well, seems weird to restrict it to only group dms

**Comment by KingRainbow44** [comment]
> I feel like this should be able to hide normal dms as well

I felt that the native Discord feature of closing DMs was comparable to this, which is why I only added support for group DMs.

**Comment by MrDiamondDog** [comment]
> > I feel like this should be able to hide normal dms as well
> 
> 
> 
> I felt that the native Discord feature of closing DMs was comparable to this, which is why I only added support for group DMs.

Ngl I forgot you could do that, ignore that then

File: src/plugins/hideGroups/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    description: "Allows hiding group DMs.",
+    settings,
+
+    startAt: StartAt.DOMContentLoaded,
+    patches: [
+        {
+            find: 'location:"private_channel"',
+            replacement: {
+                match: /return .\.isMultiUserDM\(\)/,
+                replace: (match) => {
```
Comment (MrDiamondDog) [inline src/plugins/hideGroups/index.tsx:111]: You can use capturing groups and \i to avoid finding the variable in the match string

**Comment by programminglaboratorys** [comment]
it would be cool if it automatically hides recently joined groups. it's annoying to suddenly see 300 notifications from a group you didn't ask to be joined in

**Comment by programminglaboratorys** [comment]
> it would be cool if it automatically hides recently joined groups. it's annoying to suddenly see 300 notifications from a group you didn't ask to be joined in

I understood the purpose of the plugin, quite confused of how this could be useful when it only hides dm groups in the dms list

**Comment by KingRainbow44** [comment]
> quite confused of how this could be useful when it only hides dm groups in the dms list

This plugin is meant to "add" the functionality of closing DMs for groups, not necessarily "hide" group DMs outright. It's useful for people (like me) who are in many group DMs and want to remove the clutter.

**Comment by programminglaboratorys** [comment]
> This plugin is meant to "add" the functionality of closing DMs for groups

fair point

**Comment by KingRainbow44** [comment]
never do commit shenanigans ever again

**Pull request closed without merging**